wallet-cli/src/command/blockchain/mod.rs

```diff
@@ -19,123 +19,6 @@ mod find_address;
-    println!("Network TIP is       : {}", network_tip);
-    println!("Network TIP slotid   : {}", network_slotid);
-
-    // start from our tip towards network tip
-    /*
-    if &network_tip == &our_tip {
-        println!("Qapla ! already synchronised");
-        return ();
-    }
-    */
-
-    // find the earliest epoch we know about starting from network_slotid
-    let (latest_known_epoch_id, mstart_hash, prev_hash) = match find_earliest_epoch(&storage, net_cfg.epoch_start, network_slotid.get_epochid()) {
-        None => { (net_cfg.epoch_start, Some(net_cfg.genesis.clone()), net_cfg.genesis_prev.clone()) },
-        Some((found_epoch_id, packhash)) => { (found_epoch_id + 1, None, get_last_blockid(&storage.config, &packhash).unwrap()) }
-    };
-    println!("latest known epoch {} hash={:?}", latest_known_epoch_id, mstart_hash);
-
-    let mut download_epoch_id = latest_known_epoch_id;
-    let mut download_prev_hash = prev_hash.clone();
-    let mut download_start_hash = mstart_hash.or(Some(prev_hash)).unwrap();
-
-    while download_epoch_id < network_slotid.get_epochid() {
-        println!("downloading epoch {} {}", download_epoch_id, download_start_hash);
-        let fep = FetchEpochParams {
-            epoch_id: download_epoch_id,
-            start_header_hash: download_start_hash,
-            previous_header_hash: download_prev_hash,
-            upper_bound_hash: network_tip.clone()
-        };
-        let result = net.fetch_epoch(&net_cfg, &mut storage, fep).unwrap();
-        download_prev_hash = result.last_header_hash.clone();
-        download_start_hash = result.next_epoch_hash.unwrap_or(result.last_header_hash);
-        download_epoch_id += 1;
-    }
-}
-
-fn net_sync_faster(network: String, mut storage: Storage) {
-    let netcfg_file = storage.config.get_config_file();
-    let net_cfg = net::Config::from_file(&netcfg_file).expect("no network config present");
-    let mut net = get_http_peer(network, &net_cfg);
-
-    //let mut our_tip = tag::read_hash(&storage, &"TIP".to_string()).unwrap_or(genesis.clone());
-
-    println!("Configured genesis   : {}", net_cfg.genesis);
-    println!("Configured genesis-1 : {}", net_cfg.genesis_prev);
-
-    // find the earliest epoch we know about starting from network_slotid
-    let (latest_known_epoch_id, mstart_hash, prev_hash) = match find_earliest_epoch(&storage, net_cfg.epoch_start, 100) { // TODO
-        None => { (net_cfg.epoch_start, Some(net_cfg.genesis.clone()), net_cfg.genesis_prev.clone()) },
-        Some((found_epoch_id, packhash)) => { (found_epoch_id + 1, None, get_last_blockid(&storage.config, &packhash).unwrap()) }
-    };
-    println!("latest known epoch {} hash={:?}", latest_known_epoch_id, mstart_hash);
-
-    let mut download_epoch_id = latest_known_epoch_id;
-    let mut download_prev_hash = prev_hash.clone();
-    let mut download_start_hash = mstart_hash.or(Some(prev_hash)).unwrap();
-
-    while download_epoch_id < 46 {
```

Comment:
Why is `net_sync_faster` limited to downloading epochs with ids less than magic number `46`?

Reply:
because we still miss a way to know when to stop : i.e. what is the last tip from hermes. I believe it is missing. 46 was the known last EPOCH with had packed at the time.